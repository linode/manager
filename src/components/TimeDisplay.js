import _ from 'lodash';
import moment from 'moment-timezone';
import React, { PropTypes } from 'react';

import { getStorage } from '~/storage';


export default function TimeDisplay(props) {
  const timezone = getStorage('profile/timezone') || 'UTC';
  const utcTime = moment.utc(props.time, moment.iso_8601).tz(timezone);
  const aLongTimeFromNow = moment.utc(new Date()).tz(timezone).add(100, 'year');
  const label = utcTime > aLongTimeFromNow ? 'never' : utcTime.fromNow();

  return (
    <span
      title={utcTime.format('MMM D YYYY h:mm A z')}
    >{props.capitalize ? _.capitalize(label) : label}</span>
  );
}

TimeDisplay.propTypes = {
  time: PropTypes.string,
  capitalize: PropTypes.bool,
};
