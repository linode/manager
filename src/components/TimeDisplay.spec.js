import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import TimeDisplay from '~/components/TimeDisplay';

describe('components/TimeDisplay', () => {
  it('renders properly', () => {
    const timeIso = '2017-03-30T16:48:10';
    const tDisplay = shallow(<TimeDisplay time={timeIso} />);
    const timeLong = moment.utc(timeIso, moment.ISO_8601).format('MMM D YYYY h:mm A z');
    const timeRelative = moment.utc(timeIso, moment.ISO_8601).fromNow();

    expect(tDisplay.html()).toBe(`<span title="${timeLong}">${timeRelative}</span>`);
  });
});

