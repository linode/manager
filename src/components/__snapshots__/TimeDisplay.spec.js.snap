// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TimeDisplay should render without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimeDisplay
    time="2017-03-30T16:48:10"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "10 months ago",
      "title": "Mar 30 2017 4:48 PM UTC",
    },
    "ref": null,
    "rendered": "10 months ago",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "10 months ago",
        "title": "Mar 30 2017 4:48 PM UTC",
      },
      "ref": null,
      "rendered": "10 months ago",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
